diff --git a/cardano-api.cabal b/cardano-api.cabal
index 6746e30c2..d94794832 100644
--- a/cardano-api.cabal
+++ b/cardano-api.cabal
@@ -205,12 +205,16 @@ library
     Cardano.Api.Certificate.Internal.DRepMetadata
     Cardano.Api.Certificate.Internal.OperationalCertificate
     Cardano.Api.Certificate.Internal.StakePoolMetadata
+  exposed-modules:
     Cardano.Api.Consensus.Internal.InMode
+  other-modules:
     Cardano.Api.Consensus.Internal.Mode
     Cardano.Api.Consensus.Internal.Protocol
     Cardano.Api.Consensus.Internal.Reexport
     Cardano.Api.Era.Internal.Case
+  exposed-modules:
     Cardano.Api.Era.Internal.Core
+  other-modules:
     Cardano.Api.Era.Internal.Eon.AllegraEraOnwards
     Cardano.Api.Era.Internal.Eon.AlonzoEraOnwards
     Cardano.Api.Era.Internal.Eon.BabbageEraOnwards
@@ -252,7 +256,9 @@ library
     Cardano.Api.IO.Internal.Compat.Posix
     Cardano.Api.IO.Internal.Compat.Wasm
     Cardano.Api.IO.Internal.Compat.Win32
+  exposed-modules:
     Cardano.Api.Internal.Orphans
+  other-modules:
     Cardano.Api.Internal.Orphans.Misc
     Cardano.Api.Internal.Orphans.Serialisation
     Cardano.Api.Internal.Utils
