diff --git a/nix/haskell.nix b/nix/haskell.nix
index ec137f412..cdeb9877b 100644
--- a/nix/haskell.nix
+++ b/nix/haskell.nix
@@ -110,6 +110,11 @@ let
       #
       modules =
         [
+          ({ lib, pkgs, ... }: {
+            packages.cardano-ledger-core.patches = [
+              ./cardano-ledger-core--Arbitrary-PoolMetadata.diff
+            ];
+          })
           ({ lib, pkgs, ... }: {
             packages.cardano-tracer.package.buildable = with pkgs.stdenv.hostPlatform; lib.mkForce (!isMusl);
             packages.cardano-node-chairman.components.tests.chairman-tests.buildable = lib.mkForce pkgs.stdenv.hostPlatform.isUnix;
