diff --git a/testlib/Test/Cardano/Ledger/Core/Arbitrary.hs b/testlib/Test/Cardano/Ledger/Core/Arbitrary.hs
index 0da03786f..0aad0f7c8 100644
--- a/testlib/Test/Cardano/Ledger/Core/Arbitrary.hs
+++ b/testlib/Test/Cardano/Ledger/Core/Arbitrary.hs
@@ -482,7 +482,7 @@ instance Arbitrary PoolParams where
       <*> arbitrary
 
 instance Arbitrary PoolMetadata where
-  arbitrary = PoolMetadata <$> arbitrary <*> arbitrary
+  arbitrary = PoolMetadata <$> arbitrary <*> genByteString 32
 
 instance Arbitrary StakePoolRelay where
   arbitrary = genericArbitraryU
